### PR TITLE
Fix publish package errors

### LINKED ISSUE #1012: Fix manifest error when publishing package
# 🐛 Summary #

What's wrong? Please be specific.  The manual publish package job is failing.

See this run:  https://github.com/cisagov/ScubaGear/actions/runs/8333480409/job/22805035993

## To reproduce ##

1. Run publish package with workflow dispatch.
2. Set pre-release to true and create a release name.

## Expected behavior ##

I expect it to publish the package to PSGallery without errors.

### PR DESCRIPTION
**Pull request by james-garriss**
## 🗣 Description ##

This PR fixed 2 publishing errors, one when publishing to the private repo (for testing) and one when publishing to the public repo, PSGallery.

Closes #1012

## 💭 Motivation and context ##

I fixed the bugs, then I split the single workflow into 2 workflows to separate concerns.
Along the way, I realized that the CI/CD pipeline needs to run every test every time, because some errors were not being discovered, so I updated the pipeline while I was here.
I also added a lot of debugging code in DeployUtils, as these errors were hard to find.
I also updated the PowerShell lint tests to allow Write-Host, as this is needed by the workflows to debug.

## 🧪 Testing ##

Ran pipeline.
Test both workflows.  
Added several new "alpha" releases to PSGallery (that need to be deleted).

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] PR targets the correct parent branch (e.g., main or release-name) for merge.
- [ ] Changes are limited to a single goal - *eschew scope creep!*

Yeah, no, I allowed scope creep here, because I needed other things to work in order to fix the bugs.  Sorry!

- [x] Changes are sized such that they do not touch excessive number of files.
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] These code changes follow the ScubaGear [content style guide](https://github.com/cisagov/ScubaGear/blob/main/CONTENTSTYLEGUIDE.md).
- [x] Related issues these changes resolve are linked preferably via [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).
- [x] All relevant type-of-change labels added.
- [x] All relevant project fields are set.
- [x] All relevant repo and/or project documentation updated to reflect these changes.
- [x] Unit tests added/updated to cover PowerShell and Rego changes.
- [x] All relevant functional tests passed.
- [x] All automated checks (e.g., linting, static analysis, unit/smoke tests) passed.

OPA unit tests are failing.  That's not me.

## ✅ Pre-merge checklist ##

- [x] PR passed smoke test check.
- [x] Feature branch has been rebased against changes from parent branch, as needed
- [x] Notified merge coordinator that PR is ready for merge via comment mention

## ✅ Post-merge checklist ##

- [ ] Feature branch deleted after merge to clean up repository.
- [ ] Verified that all checks pass on parent branch (e.g., main or release-name) after merge.

### CODE REVIEW

**Comment by james-garriss** [comment]
As a note, the publish to public is not going to work, because it's manual and requires workflow dispatch.  This cannot be done until pushed to main because GitHub.  I ran manual tests by hardcoding values, but to actually use it requires the PR to happen first.  Yeah, that's a chicken and egg problem.

File: .github/workflows/publish_public_package.yaml
Comment (Sloane4) [inline .github/workflows/publish_public_package.yaml:61]: This seems to be a duplicate of the jobs in publish_private_package

**Review by Sloane4 — COMMENTED** [review]
I checked main & the unit test are running without error, I suggest rebasing.

File: .github/workflows/publish_public_package.yaml
Comment (james-garriss) [reply .github/workflows/publish_public_package.yaml:61]: There are overlapping steps in the public and private workflows, but the overall jobs are different.  These were originally in a single workflow with a common first job and then "if" conditions on the public and private jobs.  The problem is that the common job always ran, even when the "if" conditions were not met.  This gave the appearance that publishing ran and worked, even when publishing never occurred.  I split these to avoid this misleading appearance.

Comment (james-garriss) [reply .github/workflows/publish_public_package.yaml:61]: Yes, I will rebase once the approvers approve.

File: Testing/Linting/MegaLinter/.powershell-psscriptanalyzer.psd1
Comment (james-garriss) [reply Testing/Linting/MegaLinter/.powershell-psscriptanalyzer.psd1:8]: Other methods of output, including Write-Output and Write-Information, do not actually write any output from the Deploy Utils script when running as part of a workflow to the console.  Write-Host does.

I will update the linter accordingly.  I did not add 'PSUseOutputTypeCorrectly'.

File: utils/DeployUtils.ps1
Comment (james-garriss) [reply utils/DeployUtils.ps1:436]: Done.

File: utils/DeployUtils.ps1
Comment (james-garriss) [reply utils/DeployUtils.ps1:184]: See above.

File: .github/workflows/publish_private_package.yaml
Comment (james-garriss) [reply .github/workflows/publish_private_package.yaml]: They are probably helpful, just in case someone else edits DeployUtils, but I commented them out and added a comment that they were helpful for debugging.

File: .github/workflows/publish_private_package.yaml
Comment (james-garriss) [reply .github/workflows/publish_private_package.yaml:62]: Added new issue, b/c this affects more than just publishing packages:
https://github.com/cisagov/ScubaGear/issues/1048

File: .github/workflows/publish_public_package.yaml
Comment (james-garriss) [reply .github/workflows/publish_public_package.yaml]: They are probably helpful, just in case someone else edits DeployUtils, but I commented them out and added a comment that they were helpful for debugging.

File: .github/workflows/publish_public_package.yaml
Comment (james-garriss) [reply .github/workflows/publish_public_package.yaml]: Done

File: .github/workflows/publish_public_package.yaml
Comment (james-garriss) [reply .github/workflows/publish_public_package.yaml:49]: See above.

**Review by Sloane4 — APPROVED** [review]
Don't forget to rebase!

**Comment by james-garriss** [comment]
@nanda-katikaneni , I think this PR is finally ready.

**Pull request merged**
